source/location_access/locacc_water_temple.cpp

```diff
@@ -245,8 +245,8 @@ void AreaTable_Init_WaterTemple() {
 
   areaTable[WATER_TEMPLE_DARK_LINK_ROOM] = Area("Water Temple Dark Link Room", "Water Temple", WATER_TEMPLE, NO_DAY_NIGHT_CYCLE, {}, {}, {
                   //Exits
-                  Entrance(WATER_TEMPLE_DRAGON_PILLARS_ROOM, {[]{return CanUse(KOKIRI_SWORD) || CanUse(MASTER_SWORD) || CanUse(BIGGORON_SWORD);}}),
-                  Entrance(WATER_TEMPLE_LONGSHOT_ROOM,       {[]{return CanUse(KOKIRI_SWORD) || CanUse(MASTER_SWORD) || CanUse(BIGGORON_SWORD);}}),
+                  Entrance(WATER_TEMPLE_DRAGON_PILLARS_ROOM, {[]{return (CanUse(KOKIRI_SWORD) || CanUse(MASTER_SWORD) || CanUse(BIGGORON_SWORD)) && Hearts > 0;}}),
+                  Entrance(WATER_TEMPLE_LONGSHOT_ROOM,       {[]{return (CanUse(KOKIRI_SWORD) || CanUse(MASTER_SWORD) || CanUse(BIGGORON_SWORD)) && Hearts > 0;}}),
```

Comment:
What happens if you go to Dark Link with 0 hearts?

Reply:
Nessy told me about this, Dark Link is invincible if you start the fight with 0 hearts.

Here's the health check from OoT decomp: https://github.com/zeldaret/oot/blob/20b90a633b18acef8bc75128c3cdf5fd05f958e1/src/overlays/actors/ovl_En_Torch2/z_en_torch2.c#L570